arrow-array/src/ffi.rs

```diff
@@ -635,6 +632,31 @@ mod tests_to_then_from_ffi {
     }
     // case with nulls is tested in the docs, through the example on this module.
 
+    #[test]
+    fn test_null_count_handling() {
+        let int32_data = ArrayData::builder(DataType::Int32)
+            .len(10)
+            .add_buffer(Buffer::from_slice_ref([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]))
+            .build()
+            .unwrap();
+        let mut ffi_array = FFI_ArrowArray::new(&int32_data);
+        assert_eq!(Some(0), ffi_array.null_count());
+        // Simulating uninitialized state
+        ffi_array.set_null_count(-1);
+        assert_eq!(None, ffi_array.null_count());
+        let int32_data = unsafe { from_ffi_and_data_type(ffi_array, DataType::Int32) }.unwrap();
+        assert_eq!(0, int32_data.null_count());
+
+        let null_data = &ArrayData::new_null(&DataType::Null, 10);
+        let mut ffi_array = FFI_ArrowArray::new(null_data);
+        assert_eq!(Some(10), ffi_array.null_count());
+        // Simulating uninitialized state
+        ffi_array.set_null_count(-1);
+        assert_eq!(None, ffi_array.null_count());
+        let null_data = unsafe { from_ffi_and_data_type(ffi_array, DataType::Null) }.unwrap();
+        assert_eq!(10, null_data.null_count());
```

Comment:
```suggestion
        assert_eq!(0, null_data.null_count());
```
The null count for a NullArray should be 0, I know it is weird...